tests/unit/multi-currency/test-class-rest-controller.php

```diff
@@ -0,0 +1,186 @@
+
+		// Act: Get the store currencies.
+		$response = $this->controller->get_store_currencies();
+
+		// Assert: Confirm the response is what we expected.
+		$this->assertEquals( $expected, $response );
+	}
+
+	/**
+	 * This test is not possible due to when MC is initialized it attempts to get the available currencies based on the
+	 * WCPay server connection and account currencies. If there is not a valid JP connection, it just returns the store currency.
+	 */
+	public function test_update_enabled_currencies_updates_currencies() {
+		$this->markTestSkipped( 'This test is not able to be completed unless local setups are refactored.' );
+		// Arrange: Set currencies for test.
+		$enabled_currencies = WC_Payments_Multi_Currency()->get_enabled_currencies();
+		$new_currencies     = [ 'USD', 'EUR', 'GBP' ];
+
+		// Arrange: Create expected response.
+		update_option( 'wcpay_multi_currency_enabled_currencies', $new_currencies );
+		$expected = rest_ensure_response( WC_Payments_Multi_Currency()->get_store_currencies() );
+
+		// Arrange: Reset the enabled currencies.
+		update_option( 'wcpay_multi_currency_enabled_currencies', $enabled_currencies );
+
+		// Arrange: Create the new REST request.
+		$request = new WP_REST_Request( 'POST', self::ROUTE . '/update-enabled-currencies' );
+		$request->set_body_params(
+			[
+				'enabled' => $new_currencies,
+			]
+		);
+
+		// Act: Update the enabled currencies.
+		$response = $this->controller->update_enabled_currencies( $request );
+
+		// Assert: Confirm the response is what we expected.
+		$this->assertEquals( $expected, $response );
+	}
+
+	public function test_get_single_currency_settings() {
+		// Arrange: Add the single currency settings.
+		update_option( 'wcpay_multi_currency_exchange_rate_cad', 'manual' );
+		update_option( 'wcpay_multi_currency_manual_rate_cad', 2 );
+		update_option( 'wcpay_multi_currency_price_rounding_cad', 1 );
+		update_option( 'wcpay_multi_currency_price_charm_cad', 0 );
+
+		// Arrange: Create expected response.
+		$expected = rest_ensure_response( WC_Payments_Multi_Currency()->get_single_currency_settings( 'CAD' ) );
+
+		// Arrange: Create the new REST request.
+		$request = new WP_REST_Request( 'GET', self::ROUTE . '/currencies/CAD' );
+		$request->set_query_params(
+			[
+				'currency_code' => 'CAD',
+			]
+		);
+
+		// Act: Get the single currency settings.
+		$response = $this->controller->get_single_currency_settings( $request );
+
+		// Assert: Confirm the response is what we expected.
+		$this->assertEquals( $expected, $response );
+	}
+
+	/**
+	 * This test uses USD due to that's the only currency available to test with.
+	 * This is due to when MC is initialized it attempts to get the available currencies based on the WCPay server
+	 * connection and account currencies. If there is not a valid JP connection, it just returns the store currency.
+	 */
+	public function test_update_single_currency_settings() {
```

Comment:
Maybe we can add an unhappy path for this test.

Reply:
Also for the other tests, to see the validation works.

Reply:
I updated the test to somewhat accommodate for the use case, deletes the option from the database, then allows the method to add it back, and checks that. This was done in https://github.com/Automattic/woocommerce-payments/pull/6468/commits/b95cb5dcaede7900d086aac1daeae14440343dc9